sdk/containerregistry/azure-containerregistry/samples/sample_upload_download_manifest.py

```diff
@@ -48,33 +42,50 @@ def upload_download_manifest(self):
                 "sample config": "content",
             }).encode())
         with ContainerRegistryClient(self.endpoint, self.credential, audience=self.audience) as client:
-            
-            
             # Upload a layer
-            layer_digest = client.upload_blob(repository_name, layer)
+            layer_digest, layer_size = client.upload_blob(repository_name, layer)
             # Upload a config
-            config_digest = client.upload_blob(repository_name, config)
+            config_digest, config_size = client.upload_blob(repository_name, config)
             # Create a manifest with config and layer info
             manifest = OCIManifest(
                 config = Descriptor(
                     media_type="application/vnd.oci.image.config.v1+json",
                     digest=config_digest,
-                    size=len(config.getbuffer())
+                    size=config_size
                 ),
                 schema_version=2,
                 layers=[
                     Descriptor(
                         media_type="application/vnd.oci.image.layer.v1.tar",
                         digest=layer_digest,
-                        size=len(layer.getbuffer()),
+                        size=layer_size,
                         annotations=Annotations(name="artifact.txt")
                     )
                 ]
             )
-
+            # Upload the manifest
             digest = client.upload_manifest(repository_name, manifest)
-            download_result = client.download_manifest(repository_name, digest)
-            print(download_result.data.read())        
+
+            # Download the manifest
+            download_manifest_result = client.download_manifest(repository_name, digest)
+            downloaded_manifest = download_manifest_result.manifest
+            download_manifest_stream = download_manifest_result.data
+            print(download_manifest_stream.read())
+            # Download the layers
+            for layer in downloaded_manifest.layers:
+                downloaded_layer = client.download_blob(repository_name, layer.digest)
+                print(downloaded_layer.read())
```

Comment:
Based on the draft protocol: https://github.com/Azure/azure-sdk-for-python/pull/29265/files
The download API would work something like this.
Also - every sample that demonstrates downloading must also demonstrate handling for the digest validation.
```python
for layer in downloaded_manifest.layers:
    with client.download_blob(repository_name, layer.digest) as stream:
        print(b"".join(stream))
```
Or writing to file:
```python
for layer in downloaded_manifest.layers:
    try:
        with client.download_blob(repository_name, layer.digest) as stream:
            with open("layer_name", "wb") as layer_file:
                for data in stream:
                    layer_file.write(data)
    except ValueError:
        # Downloaded digest value did not match. Deleting file
        os.remove("layer_file")
```